src/main/scala/org/broadinstitute/dsde/firecloud/service/FireCloudDirectives.scala

```diff
@@ -35,4 +37,14 @@ trait FireCloudDirectives extends spray.routing.Directives with PerRequestCreato
       }
     }
   }
+
+  def encodeUri(path: String): String = {
+    val pattern = """(https|http)://([^/\r\n]+)(/[^\r\n]*)?""".r
```

Comment:
I'm not seeing how this replaces white space in an url. I see returns and newlines, but no \s or \t, etc. Do you have an example poorly named workspace for this?